plugins/woocommerce/includes/class-wc-webhook.php

```diff
@@ -274,21 +274,25 @@ private function is_valid_processing_action( $arg ) {
 	private function is_valid_resource( $arg ) {
 		$resource = $this->get_resource();
 
-		if ( in_array( $resource, array( 'order', 'product', 'coupon' ), true ) ) {
+		if ( in_array( $resource, array( 'product', 'coupon' ), true ) ) {
 			$status = get_post_status( absint( $arg ) );
 
 			// Ignore auto drafts for all resources.
 			if ( in_array( $status, array( 'auto-draft', 'new' ), true ) ) {
 				return false;
 			}
+		}
 
-			// Ignore standard drafts for orders.
-			if ( 'order' === $resource && 'draft' === $status ) {
+		if ( 'order' === $resource ) {
+			// Check registered order types for order types args.
+			if ( ! OrderUtil::is_order( absint( $arg ), wc_get_order_types( 'order-webhooks' ) ) ) {
 				return false;
 			}
 
-			// Check registered order types for order types args.
-			if ( 'order' === $resource && ! OrderUtil::is_order( absint( $arg ), wc_get_order_types( 'order-webhooks' ) ) ) {
+			$order = wc_get_order( absint( $arg ) );
+
+			// Ignore standard drafts for orders.
+			if ( in_array( $order->get_status(), array( 'draft', 'auto-draft', 'new' ), true ) ) {
```

Comment:
Thinking of [this discussion](https://github.com/woocommerce/woocommerce/issues/27650), do we definitely want to ignore drafts?

Reply:
I guess we should right based on the discussion, to prevent duplicate webhooks from being fired. Perhaps I am missing something?

Reply:
Ignore me 😅 ... my (mis-)reading was that we should continue to emit on creation, even if the status is draft, but of course that whole discussion referred to products, not orders.